README.md

```diff
@@ -0,0 +1,116 @@
+provides a convenient data structure, which is very fast on reads, and quite
+fast on writing. The algorithm used is very close to django-treebeard's
+materialized paths, but with all the power of PostgreSQL.
+
+The main downside of using ltree is that you have to maintain the materialized
+path yourself. It doesn't come with any tool to do it automatically.
+But fortunately, it's actually quite simple to maintain this path using
+PostgreSQL triggers!
+
+
+## Integration with Django
+
+In [`demo/categories/ltree.py`](/demo/categories/ltree.py) you will find a very
+simple Django field for the ltree data type. This field can be used in any
+Django model, and adds two lookups: `descendant` and `ancestor`. Those lookups
+allow you to query the descendants or the ancestors of any object with a very
+simple SQL query.
+
+For example, let's say you have the following model:
+```python
+from django.db import models
+
+from project.ltree import LtreeField
+
+
+class Category(models.Model):
+  parent = models.ForeignKey('self', null=True)
+  code = models.CharField(maxlength=32, unique=True)
+  path = LtreeField()
+```
+
+The `path` field represents the path from the root to the node, where each
+node is represented by its code (it could also be its id, but using the code
+is more readable when debugging). For example, if you have a `genetic`
+category, under a `science` category, under a `top` category, its path would be
+`top.science.category`.
+
+Thanks to the `descendant` and `ancestor` lookups, the `get_descendants`
+method in django-mptt can be rewritten as:
+```python
+def get_descendants(self):
+    return Category.objects.filter(path__descendant=self.path)
+```
+
+This would generate a SQL query close to:
+```sql
+SELECT * FROM category WHERE path <@ 'science.biology'
+```
+
+## The magic part: PostgreSQL triggers
+
+If you just add a ltree field to your model, you will have to keep the field
+up-to-date when inserting or updating instances.  We could do that with Django
+signals, but it turns out that PostgreSQL is far better for maintaining
+integrity & writing efficient code.
+
+Every time we insert or update a row, we can reconstruct its path by appending
+its code to the path of its parent. If the path changed, we'll also need to
+update the path of the children, which is a simple `UPDATE`.
+
+This can be done easily with [PostgreSQL
+triggers](https://www.postgresql.org/docs/current/static/sql-createtrigger.html).
+You can find an implementation of those triggers in the file
+[`demo/categories/sql/triggers.sql`](/demo/categories/sql/triggers.sql).
+
+
+## The demo
+
+In the demo, the following files are the most important:
+- [`demo/categories/models.py`](/demo/categories/models.py): The definition of
+  a model using ltree
+- [`demo/categories/ltree.py`](/demo/categories/ltree.py): A very simple Django
+  field for ltree. More lookups could be added (one for `~` for instance).
```

Comment:
Did you know : the `@>` operator has already been implemented in django as `__contains` as of Django >= 1.8 (https://github.com/django/django/blob/stable/1.8.x/django/contrib/postgres/lookups.py). and `<@` as `contained_by`, so you could do the same without defining custom lookups :)

Reply:
Yes, but I find `ancestor` and `descendant` more convenient though.

Reply:
(or you could inherit from the existing lookups and just redifine their lookup_name if you really want to use "ancestor")